applications/FluidDynamicsApplication/custom_elements/fluid_element.cpp

```diff
@@ -553,6 +553,24 @@ array_1d<double, 3> FluidElement<TElementData>::GetAtCoordinate(
     return result;
 }
 
+template <class TElementData>
+Matrix FluidElement<TElementData>::GetAtCoordinate(
+    const typename TElementData::NodalTensorData& rValues,
+    const typename TElementData::ShapeFunctionsType& rN) const
+{
+    Matrix result = ZeroMatrix(Dim,Dim);
+
+    for (size_t i = 0; i < NumNodes; i++) {
+        for (size_t j = 0; j < Dim; j++) {
+            for (size_t k = 0; k < Dim; k++) {
+                result(j, k) += rN[i] * rValues[i](j, k);
```

Comment:
I think that you can avoid the `j` and `k` loops by doing `noalias(result) += rN[i] * rValues[i]`.